gh-127794: Validate header name according to rfc-5322 (#127820)
python/cpython: Misc/NEWS.d/next/Library/2024-12-11-17-44-36.gh-issue-127794.VwmRsp.rst

The :meth:`email.message.Message.add_header` method now validates header

> field names according to :rfc:`5322`.

Reviewer: This needs rewriting now, too.  Something like ":meth:email.message.Message now validates header field names according to :rfc`5322` when headers are added via `add_header` or `__setitem__` and raises a ValueError if they contain invalid characters."

> The :meth:`email.message.Message.add_header` method now validates header
> field names according to :rfc:`5322` when headers are added via add_header
> or __setitem__ and raises a ValueError if they contain invalid characters.

Reviewer 2: ```suggestion
The :meth:`email.message.Message.add_header` method now validates header
field names according to :rfc:`5322` and raises a :exc:`ValueError` if
they contain invalid characters. A similar argument applies if headers
are added through :meth:`Message.__setitem__ <object.__setitem__>`.
```

> When headers are added to :meth:`email.message.Message` objects, either through

Reviewer 2: ```suggestion
When headers are added to :class:`email.message.Message` objects, either through
```

> field names according to :rfc:`5322` and raises a :exc:`ValueError` if

Reviewer 2: Can we have the dedicated section with field names specifications? If you find it, you can link it as <code>:rfc:\`5322#section-X.Y\`</code> and this would users knowing about it instead of going through the entire RFC.

Author: Addressed in 56a77e93524

the field name is now validated according to  :rfc:`RFC 5322, Section 3.6.8 <5322#section-3.6.8>`

> and a :exc:`ValueError` is raised if the field name contains any invalid characters.

Reviewer 2: ```suggestion
When headers are added to :class:`email.message.Message` objects, either through
:meth:`email.message.Message.__setitem__` or :meth:`email.message.Message.add_header`,
the field name is now validated according to :rfc:`RFC 5322, Section 2.2 <5322#section-2.2>`
and a :exc:`ValueError` is raised if the field name contains any invalid characters.
```

Actually, the definition of a header field name is in §2.2. 3.6.8 was only for optional fields but the requirements are identical for the name (2.2 says "all characters from 33 to 126 except colon and 3.6.8 says all printable characters except space and colon, but space is 32 so it's the same).

The :meth:`email.message.Message.add_header` method now validates header

> field names according to :rfc:`RFC 5322 section 3.6.8 <5322#section-3.6.8>`

Reviewer 2: ```suggestion
field names according to :rfc:`RFC 5322, Section 3.6.8 <5322#section-3.6.8>`
```

How does

```rst
:rfc:`5322#section-3.6.8`
```

renders? it should be picked up automatically by Sphinx but I don't remember how it's being rendered. If it's rendered fine, we can keep a shorter formulation.

and raises a :exc:`ValueError` if they contain invalid characters. A similar
argument applies if headers are added through :meth:`Message.__setitem__

> <object.__setitem__>`.

Reviewer: I would phrase this as follows:

```
When headers are added to :meth:`email.message.Message` objects, either through
:meth:`email.message.Message.__setitem__` or :meth:`email.message.Message.add_header`,
the field name is now validated according to  :rfc:`RFC 5322, Section 3.6.8 <5322#section-3.6.8>`
and a :exc:`ValueError` is raised if the field name contains any invalid characters.
```
